### PR TITLE
Add basic Polynomial Chaos Expansion

### PR DESCRIPTION
**Pull request by FriesischScott**
Can compute PCE using least squares minimization or full quadrature rules using legendre basis functions.

The code should be fully extendible to other basis function, e.g. hermite by overloading a few methods in `pcebases.jl`.

Unfortunately `SparseGrids.jl` doesn't seem to support mixing nodes/weights for different families of polynomials. Having sparse quadrature rules will be very important going forward.

I'll add hermite polynomials before merging this.

### CODE REVIEW

**Comment by FriesischScott** [comment]
I need to find a few more examples to check hermite polynomials and also hybrid bases.

**Comment by FriesischScott** [comment]
> Looks great to me, a very nice contribution and implementation. Only (minor) suggestion is that evaluate! could allow for raw samples as well as a DF. But we would have to change this for the evaluate functions

Maybe create an issue for it so we can discuss this generally?

**Review by AnderGray — APPROVED** [review]
Looks good to me, sorry for taking so long to get back to you

File: demo/metamodels/polynomialchaosexpansion.jl [hunk 1]
```diff
@@ -28,17 +28,18 @@ println("GQ Variance: $(var(pceGQ))")
...
-sampsGQ = sample(pceGQ, 10^5)
-
-histogram(sampsLS[!,:y], alpha = 0.5, label = "Least squares", normalize = :probability, bins = 100)
-histogram!(sampsGQ[!,:y], alpha = 0.5, label = "Quadrature", normalize = :probability, bins = 100)
-histogram!(sampsMC[!,:y], alpha = 0.5, label = "Monte Carlo", normalize = :probability, bins = 100)
+samplesLS = sample(pceLS, 10^5)
+samplesGQ = sample(pceGQ, 10^5)
+
+histogram(
+    samplesLS.y; alpha=0.5, label="Least squares", normalize=:probability, bins=100
```
[the remark this answers is not in the document]
Comment (FriesischScott) [reply demo/metamodels/polynomialchaosexpansion.jl:42]: Yeah, that doesn't seem to make sense.

**Pull request merged**